fish_probes/C_predict_probes.py

```diff
@@ -89,7 +89,53 @@ def find_conserved_regions(seq_sel_clade,k,perc_seq_with_kmer):
     # return
     return kmers_recall,kmers_precision
 
+def get_kmer_sens(seq_sel_clade,kmer):
 
+    #all_strings_kmers = dict()
+    #for s in seq_sel_clade:
+    #   allK,allK_N = find_kmers(seq_sel_clade[s],len(kmer))
+    #   all_strings_kmers[s] = allK
+
+    all_kmers = set()
+    all_kmers.add(kmer)
+    #for s in all_strings_kmers:
+    #   for kmer in all_strings_kmers[s]:
+    #       all_kmers.add(kmer)
+
+    if VERBOSE > 2:
+        #UTIL_log.print_message("Identifed "+str(len(all_kmers))+" unique "+str(k)+"-mers.")
+        UTIL_log.print_message("Calculating sensitivity for k-mer {}".format(kmer))
+    # # now we count how many times it appear
+    count_mers = dict()
+    for kmer in list(all_kmers):
+        count_mers[kmer] = 0
+    # # now we add the counts per k-mer
+    #for _, kmers in all_strings_kmers.items():
+    for _, seq in seq_sel_clade.items():
+        if kmer in seq:
+            count_mers[kmer] = count_mers[kmer] + 1
```

Comment:
In the for-loop in line 114, you only ever check the last `kmer` from `list(all_kmers)`, not all. If you want to check all of `all_kmers`, then you'd have to nest the for-loop in 114 into the one in 110.

Reply:
This is the crucial review item - the rest is cosmetics and low-level tuning.